modules/vpc-endpoints/variables.tf

```diff
@@ -0,0 +1,34 @@
+variable "vpc_id" {
+  description = "The ID of the VPC in which the endpoint will be used"
+  type        = string
```

Comment:
Let's add `create` boolean flag to be able to disable this module the same way as we do in the VPC. For this reason there should be `default` in every variable.

Reply:
This is one of my favorite error messages :) objects vs objects battle...

Try writing like:

```hcl
endpoints = var.create ? tomap(var.endpoints) : tomap({})
```

Ref: https://github.com/terraform-aws-modules/terraform-aws-route53/blob/master/modules/zones/main.tf#L2

Reply:
Yes, this is the best way. :)